ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/Uniform.hs

```diff
@@ -105,23 +108,27 @@ makeProperty genesisTest advCount StateView {svSelectedChain} killed =
       [] -> "No peers were killed"
       peers -> "Some peers were killed: " ++ intercalate ", " (condense <$> peers)
 
-    honestTipSlot = AF.headSlot (btTrunk gtBlockTree)
+    honestTipSlot = At $ tbSlot $ snd $ last $ mapMaybe fromBlockPoint $ value $ honest schedule
```

Comment:
This is a very elaborated line; makes me question whether we have the right representations or helpers. Maybe we'll see things emerge from such uses!